shared/src/main/scala/tastyquery/Contexts.scala

```diff
@@ -19,6 +21,12 @@ object Contexts {
   transparent inline def clsCtx(using clsCtx: ClassContext): ClassContext = clsCtx
   transparent inline def defn(using baseCtx: BaseContext): baseCtx.defn.type = baseCtx.defn
 
+  object permissions:
+    opaque type InitAll = Unit
```

Comment:
It seems you did it again: creating an artificial permission to "ensure" that one specific method is only called where you have the "permission" to do so, but in fact "all" (i.e., one in this case) places where you call the method give themselves the permission immediately before calling it. This is only providing the illusion of safety, without making anything any safer. A permission system like that works if it makes sense for the permissions to be transitively passed through, and where the only way to create the permission is with something that *handles* the thing it permits. (like a `try/catch` creates the permission to `throw` within the `try`).

There is no need for this. Give `initPackages` a slightly scarier name like `initAllPackagesFromBaseContext`, and you will achieve the same level of "safety".

Reply:
my intention with this was just to make `initPackages` only callable from `Contexts`, (basically simulating an non-existent access control modifier) rather than global protection against a sub process eventually calling it - I will remove it though

Reply:
removed